src/venue/core.cljs

```diff
@@ -183,21 +216,24 @@
 
 (defn- start-service-loop!
   []
-  (go-loop [[caller cursor service-id req-id args] (<! service-request-ch)]
-    (if-let [service (get-in @state [:services service-id])]
-      (let [c (chan)
-            to (timeout 5000)
-            rfn (fn [caller outcome req-id data cursor]
-                  (when (satisfies? IHandleResponse caller)
-                    (handle-response caller outcome req-id data cursor)))]
-        (log-debug "Received service request:" service-id req-id)
-        (go
-          (alt!
-            c  ([[outcome data]] (rfn caller outcome req-id data cursor))
-            to ([_] (rfn caller :failure req-id "The service request timed out" cursor))))
-        (service req-id args c))
-      (log-severe "A request was sent to an unknown service: " service-id))
-    (recur (<! service-request-ch))))
+  (go-loop []
+    (let [[caller cursor service-id req-id args] (<! service-request-ch)]
```

Comment:
I'm guessing the messages pushed onto service-request-ch are going to be vectors of [caller cursor service-id req-id args]. Should these be maps instead so we don't care about order or extra keys?